engine/src/conversion/analysis/struct_accessors.rs

```diff
@@ -0,0 +1,142 @@
+                            fun: Box::new(FuncToConvert {
+                                provenance: Provenance::SynthesizedOther,
+                                ident,
+                                doc_attrs: Vec::new(),
+                                inputs: [fnarg].into_iter().collect(),
+                                variadic: false,
+                                output: parse_quote! {
+                                    -> #field_type
+                                },
+                                vis: parse_quote! { pub },
+                                virtualness: crate::conversion::api::Virtualness::None,
+                                cpp_vis: crate::conversion::api::CppVisibility::Public,
+                                special_member: Some(SpecialMemberKind::GeneratedAccessor),
+                                unused_template_param: false,
+                                references: Default::default(),
+                                original_name: None,
+                                self_ty: None,
+                                synthesized_this_type: None,
+                                add_to_trait: None,
+                                synthetic_cpp: Some((body, CppFunctionKind::Function)),
+                                is_deleted: false,
+                            }),
+                            analysis: (),
+                        })
+                    }
+                }
+                // TODO: is it even possible for these to occur?
+                syn::Fields::Unnamed(_) => todo!(),
+                syn::Fields::Unit => todo!(),
+            };
+
+            // Generate the accessors (if any) + the struct itself
+            Ok(Box::new(accessors.into_iter().chain(std::iter::once(
+                Api::Struct {
+                    name: struct_name,
+                    details,
+                    analysis,
+                },
+            ))))
+        },
+        Api::enum_unchanged,
+        Api::typedef_unchanged,
+    );
+
+    results
+}
+
+fn should_generate_accessor(field: &Field) -> bool {
+    // Don't generate accessors for non-public fields
+    if !matches!(field.vis, Visibility::Public(_)) {
+        return false;
+    }
+
+    // Don't generate accessors for cpp reference fields
+    // TODO: is this restriction necessary?
+    if field.attrs.iter().any(is_cpp_reference) {
+        return false;
+    }
+
+    // Don't generate accessors for "fake" bindgen fields which wouldn't appear directly in the C++ struct
+    // TODO: is there a better way to tell which fields these are?
+    let field_name = field.ident.as_ref().unwrap().to_string();
+    if field_name == "vtable_" || field_name == "_address" || field_name == "_base" {
+        return false;
+    }
+
+    return true;
+}
+
+fn is_cpp_reference(attr: &syn::Attribute) -> bool {
+    return *attr == parse_quote! { #[cpp_semantics(reference)] }
```

Comment:
Please could you add logic to `src/conversion/parse/bindgen_semantic_attributes.rs` to understand this attribute, then use that to do this attribute parsing? (It's fine with me, for now, if that code needs to be made visible outside `parse`)

Reply:
Just to confirm, did you mean you are ok with that method being public as it is here: https://github.com/google/autocxx/pull/1133/files#diff-887ed0eb29c1743994a938da92e9c5455536ef703017656b53b7f23417f77f18R79?